src/main/java/com/adyen/model/checkout/AbstractPaymentsResponse.java

```diff
@@ -479,5 +503,70 @@ public Date getExpiryDate() {
         return DateUtil.parseMYDate(expiryDate);
     }
 
+    /**
+     * The result of the payment. Possible values:
+     * * **AuthenticationFinished** – The payment has been successfully authenticated with 3D Secure 2. Returned for 3D Secure 2 authentication-only transactions.
+     * * **Authorised** – The payment was successfully authorised. This state serves as an indicator to proceed with the delivery of goods and services. This is a final state.
+     * * **Cancelled** – Indicates the payment has been cancelled (either by the shopper or the merchant) before processing was completed. This is a final state.
+     * * **ChallengeShopper** – The issuer requires further shopper interaction before the payment can be authenticated. Returned for 3D Secure 2 transactions.
+     * * **Error** – There was an error when the payment was being processed. The reason is given in the `refusalReason` field. This is a final state.
+     * * **IdentifyShopper** – The issuer requires the shopper's device fingerprint before the payment can be authenticated. Returned for 3D Secure 2 transactions.
+     * * **Pending** – Indicates that it is not possible to obtain the final status of the payment. This can happen if the systems providing final status information for the payment are unavailable, or if the shopper needs to take further action to complete the payment. For more information on handling a pending payment, refer to [Payments with pending status](https://docs.adyen.com/development-resources/payments-with-pending-status).
+     * * **Received** – Indicates the payment has successfully been received by Adyen, and will be processed. This is the initial state for all payments.
+     * * **RedirectShopper** – Indicates the shopper should be redirected to an external web page or app to complete the authorisation.
+     * * **Refused** – Indicates the payment was refused. The reason is given in the &#x60;refusalReason&#x60; field. This is a final state.
+     */
+    @JsonAdapter(ResultCodeEnum.Adapter.class)
+    public enum ResultCodeEnum {
```

Comment:
I'm not super familiar with how this project organizes its classes, but there's enough in here it seems reasonable to me to move this enum to its own file.